vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc

```diff
@@ -297,54 +297,53 @@ void BatPublishers::OnPublisherInfoSaved(
   SynopsisNormalizer();
 }
 
-void BatPublishers::SetPublisherExclude(const std::string& publisher_id,
-                               const ledger::PUBLISHER_EXCLUDE& exclude) {
-    ledger_->GetPublisherInfo(publisher_id,
-        std::bind(&BatPublishers::OnSetPublisherExclude,
-                  this,
-                  exclude,
-                  _1,
-                  _2));
+void BatPublishers::SetPublisherExclude(
+    const std::string& publisher_id,
+    const ledger::PUBLISHER_EXCLUDE& exclude,
+    ledger::SetPublisherExcludeCallback callback) {
+  ledger_->GetPublisherInfo(
+    publisher_id,
+    std::bind(&BatPublishers::OnSetPublisherExclude,
+              this,
+              exclude,
+              _1,
+              _2,
+              callback));
 }
 
 void BatPublishers::OnSetPublisherExclude(
     ledger::PUBLISHER_EXCLUDE exclude,
     ledger::Result result,
-    ledger::PublisherInfoPtr publisher_info) {
+    ledger::PublisherInfoPtr publisher_info,
+    ledger::SetPublisherExcludeCallback callback) {
   if (result != ledger::Result::LEDGER_OK &&
       result != ledger::Result::NOT_FOUND) {
+    callback(ledger::Result::LEDGER_ERROR);
```

Comment:
Seems like we should just return `result` here.

Reply:
done